src/main/java/org/broadinstitute/hellbender/tools/exome/ReadCountCollectionUtils.java

```diff
@@ -312,12 +408,235 @@ public static String getSampleNameFromReadCounts(final ReadCountCollection count
                 writer.writeRecord(new ReadCountRecord.SingleSampleRecord(new Target(interval), coverage));
             }
         } catch (final IOException e) {
-            throw new UserException.CouldNotCreateOutputFile(outFile, e);
+            throw new UserException.CouldNotCreateOutputFile(outName, e);
         }
 
+    }
 
+    /**
+     * Impute zero counts to the median of non-zero values in the enclosing target row.
+     *
+     * <p>The imputation is done in-place, thus the input matrix is well be modified as a result of this call.</p>
+     *
+     * @param readCounts the input and output read-count matrix.
+     */
+    public static void imputeZeroCountsAsTargetMedians(final ReadCountCollection readCounts, final Logger logger) {
+
+        final RealMatrix counts = readCounts.counts();
+        final int targetCount = counts.getRowDimension();
+
+        final Median medianCalculator = new Median();
+        int totalCounts = counts.getColumnDimension() * counts.getRowDimension();
+
+        // Get the number of zeroes contained in the counts.
+        final long totalZeroCounts = IntStream.range(0, targetCount)
+                .mapToLong(t -> DoubleStream.of(counts.getRow(t))
+                        .filter(c -> c == 0.0).count()).sum();
+
+        // Get the median of each row, not including any entries that are zero.
+        final double[] medians = IntStream.range(0, targetCount)
+                .mapToDouble(t -> medianCalculator.evaluate(
+                        DoubleStream.of(counts.getRow(t))
+                                .filter(c -> c != 0.0)
+                                .toArray()
+                )).toArray();
+
+        // Change any zeros in the counts to the median for the row.
+        counts.walkInOptimizedOrder(new DefaultRealMatrixChangingVisitor() {
+            @Override
+            public double visit(final int row, final int column, final double value) {
+                return value != 0 ? value : medians[row];
+            }
+        });
```

Comment:
What is the reason to not do it using a stream here, and a lambda expression instead of anonymous class?